### PR TITLE
add github pages workflow and scripts

### PR DESCRIPTION
**Pull request by warm-coolguy**
## Summary

Adding scripts and files to create a basic GitHub page featuring the Snowbox and our documentation. An action to automate reconstruction has been added to update the GitHub page on merges to main. However, the latter is to be tested *by merging*.

## Instructions for local reproduction and review

Run `npm run buildPages && npx http-server ./pages_output -o index.html`

### CODE REVIEW

File: pages/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { getClientConfig } from '../viteConfigs'
+
+export default getClientConfig({
+  root: '.',
+  define: {
+    'process.env.NODE_ENV': `"${process.env.NODE_ENV}"`,
+  },
```
Comment (dopenguin) [inline pages/vite.config.js:7]:
```suggestion
```
Duplicating `getClientConfig`

File: pages/index.html [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+      client), see the programmatic information binding example below.
+    </p>
+    <label id="language-switcher-label">
+      Language in map client:
+      <select
+        id="language-switcher"
+        style="border-style: solid; background: white"
+      >
+        <option value="de">Deutsch</option>
+        <option value="en">Englisch</option>
```
Comment (dopenguin) [inline pages/index.html:103]:
```suggestion
        <option value="de">German</option>
        <option value="en">English</option>
```

File: pages/index.html [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    </label>
+    <div id="polarstern" class="polarstern"></div>
+    <h2>Example for programmatic information binding</h2>
+    <p>
+      This illustrates which kind of data can be retrieved from the map client.
+      For a full overview, please check the full documentation per plugin.
+    </p>
+    <p>Current zoom level: <span id="vuex-target-zoom" /></p>
+    <p>GFI information: <span id="vuex-target-gfi" /></p>
+    <p>PIN coordinates: <span id="vuex-target-pin-coordinate" /></p>
```
Comment (dopenguin) [inline pages/index.html:114]:
```suggestion
    <p>Pin coordinates: <span id="vuex-target-pin-coordinate" /></p>
```

File: .github/workflows/publish-pages.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
```
Comment (dopenguin) [inline .github/workflows/publish-pages.yml:15]:
```suggestion
          node-version: 18.13.0
```
`18.16.0` uses a different major version of `npm` than `18.13.0`

**Review by dopenguin — CHANGES_REQUESTED** [review]

File: pages/index.html (hunk 2 above)
Comment (warm-coolguy) [reply pages/index.html]: Resolved in https://github.com/Dataport/polar/pull/3/commits/e11868c43d959583c4e8e806f1c9bca41bc6a57f. ☎️

**Review by dopenguin — APPROVED** [review]
Looks great!

**Pull request merged**